### PR TITLE
Update to egui 0.21.0

### PR DESCRIPTION
**Pull request by emilk**
This PR updates to egui 0.21.0

I would greatly appreciate if you could merge this and publish a new version of `egui_dock` as soon as you can 🙏 

This also adds a small new feature: `TabViewer::on_tab_button`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,8 +18,11 @@ default = []
 serde = ["dep:serde", "egui/serde"]
 
 [dependencies]
-egui = "0.20"
+egui = { version = "0.21", default-features = false }
```
Comment (emilk) [inline Cargo.toml:21]: Small change here: egui has the `default_fonts` feature by default. It bundles the default egui fonts (hence the name).

Previously, `egui_dock` would implicitly "force" the `default_fonts` feature onto users of `egui_dock`, bloating their binaries if they were not using `default_fonts` themselves.

File: src/lib.rs [hunk 2]
```diff
@@ -162,6 +163,9 @@ pub trait TabViewer {
     /// The title to be displayed.
     fn title(&mut self, tab: &mut Self::Tab) -> WidgetText;
 
+    /// Called after each tab button is shown, so you can add a tooltip, check for clicks, etc.
+    fn on_tab_button(&mut self, _tab: &mut Self::Tab, _response: &egui::Response) {}
```
Comment (emilk) [inline src/lib.rs:167]: New feature!

**Comment by emilk** [comment]
This is finally ready for review @Adanos020

File: Cargo.toml (hunk 1 above)
Comment (Adanos020) [reply Cargo.toml:21]: Pardon my cargo-ignorance; if a user wanted to use the version of `egui` exported by `egui_dock`, what would they need to do to enable `default_fonts` in it?

**Review by Adanos020 — APPROVED** [review]

**Comment by emilk** [comment]
@Adanos020 usually a user would have:

```toml
egui = "0.12"
egui_dock = "0.4"
```

and it would just work

**Comment by Adanos020** [comment]
@emilk Then before publishing 0.4, I'd add a note about that in the Readme. I'll merge this PR for now and do that in a separate one.

**Pull request merged**